Framework/API/src/FileFinder.cpp

```diff
@@ -605,12 +605,31 @@ FileFinderImpl::findRuns(const std::string &hintstr,
       if (runEndNumber < runNumber) {
         throw std::invalid_argument("Malformed range of runs: " + *h);
       }
+      std::string previousPath, previousExt;
       for (int irun = runNumber; irun <= runEndNumber; ++irun) {
         run = std::to_string(irun);
         while (run.size() < nZero)
           run.insert(0, "0");
+
+        // Quick check if file can be created from previous successfully found
+        // path/extension
+        if (!previousPath.empty() && !previousExt.empty()) {
+          try {
+            const Poco::File file(previousPath + p1.first + run + previousExt);
```

Comment:
Can we not duplicate `previousPath + p1.first + run + previousExt` below? From a readability point of you I need to double take to check that they are the same.

I think `Poco::File` has a string method?